cl_manycore/libraries/bsg_manycore_cuda.cpp

```diff
@@ -5,354 +5,1089 @@
 #include <bsg_manycore_elf.h>
 #include <bsg_manycore_mem.h>
 #include <bsg_manycore_loader_dep.h>
+#include <bsg_manycore_loader.h>
+#include <bsg_manycore.h>
+#include <bsg_manycore_printing.h>
+#include <bsg_manycore_eva.h>
+#include <bsg_manycore_origin_eva_map.h>
 
-static const uint32_t KERNEL_REG = 0x1000 >> 2; //!< EPA of kernel. 
-static const uint32_t ARGC_REG = 0x1004 >> 2; //!< EPA of number of arguments kernel will use. 
-static const uint32_t ARGV_REG = 0x1008 >> 2; //!< EPA of arguments for kernel. 
-static const uint32_t SIGNAL_REG = 0x100c >> 2; //!< EPA of register that holds signal address. Tile will write to this address once it completes the kernel.   
+static const hb_mc_epa_t KERNEL_REG = 0x1000; //!< EPA of kernel. 
```

Comment:
Can you define these as Macros instead of static const variables?

Reply:
And can you make the naming scheme similar to other defined addresses? See mmio.h, where I think the standard is HB_MC_MMIO_<NAME>_ADDR

Reply:
Good point! I converted them to macro defines, changed the name in accordance with the naming schemes, and moved them to the header file. 
The plan is to remove them entirely and load these values from the binary file (issue #175 ), I looked into this a bit but its a bit complicated, since these values are being used in the riscv assembly and I still haven't found a way to integrate defined macros or variables into the assembly. I will address this issue in a separate pull request.

Reply:
We can look into it! Maybe I don't understand the purpose.

The most future proof would be to define a getter function that takes a binary and returns the EVA (?) and the calling EVA to NPA to get the EPA. 

But the macros are good for now.

Reply:
Yes the problem isn't on the host side I can handle that, the problem is that I have to define these variables on the device code so they show up in the elf as symbols, and to use the variables in the assembly code on the device side (I can't integrate variables or even a define macro into riscv assembly code, I've been using magic numbers for now), that's the thing I have to figure out.